llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -2034,6 +2080,202 @@ PackOp::reifyResultShapes(OpBuilder &builder,
+        // dims_pos = [0, 1] and inner_tiles = [8, 2]. This is rejected today as
+        // 8 and 2 do not fully divide 13 and 15. But this is actually a legit
+        // operation. From the 16x16 we can extract 13x15 discarding previously
+        // introduced padding values. We need to guard the extraction with an
+        // if.
+        int64_t sizeDim = inputType.getDimSize(dim) * tile;
+        inferredShape.push_back(sizeDim);
+      }
+    } else {
+      inferredShape.push_back(inputType.getShape()[dim]);
+    }
+  }
+  assert(inferredShape.size() == getOutputRank() &&
+         "expect inferredShape to match output rank");
+  return TypeSwitch<Type, ShapedType>(inputType)
+      .Case<RankedTensorType>([&](RankedTensorType t) -> ShapedType {
+        return RankedTensorType::get(inferredShape, inputType.getElementType());
+      })
+      .Case<MemRefType>([&](MemRefType t) -> ShapedType {
+        return MemRefType::get(inferredShape, inputType.getElementType());
+      })
+      .Default([&](Type t) {
+        llvm_unreachable("unexpected type");
+        return nullptr;
+      });
+}
+
+LogicalResult UnPackOp::verify() {
+  Operation *op = getOperation();
+  size_t numberOfBlockingFactors = getMixedTiles().size();
+  SmallVector<int64_t> dimsPos = extractFromI64ArrayAttr(getDimsPos());
+  if (failed(commonVerifierPackAndUnPackOp(*this))) {
+    return failure();
+  }
+  // Blocking factors must be less or equal than the input rank, and must
+  // match the number of `dims_pos`.
+  if (numberOfBlockingFactors > getOutputRank()) {
+    return op->emitError(
+        "blocking factors must be less or equal than the output rank");
+  }
+  // Require `dim_pos` to be in-bound. `dim_pos` carries the index of the
+  // dimensions to block.
+  if (!isInBound(dimsPos, getOutputRank()))
+    return op->emitError("out-of-bound position");
+  // Require input rank to match output rank + number of blocking factors.
+  if ((getOutputRank() + numberOfBlockingFactors) != getInputRank()) {
+    return op->emitError(
+        "input rank must equal output rank + blocking factors");
+  }
+  // Bail out if the tile does not divide the dimension fully. In the case of
+  // dynamic tile factors or dimensions, having a partial tile is undefined
+  // behavior. TODO: (lorenzo): We could relax this if we allow to `undo` the
+  // padding done in the pack operation. The product of dim(point_loop) and
+  // dim(tile_loop) >= dim(input).
+  if (areNotFullTiles(getOutputShape(), getDimAndTileMapping())) {
+    return op->emitError(
+        "invalid tile factor provided. Only full tiles are supported");
+  }
+  // Verify result type against inferred type.
+  ShapedType expectedType = inferResultType();
+  if (expectedType != getOutputType()) {
+    return op->emitError(
+               "inferred type do not match provied output type. Expected ")
+           << expectedType << " but got: " << getOutputType();
+  }
+  return success();
+}
```

Comment:
@MaheshRavishankar landed a [fix](https://github.com/iree-org/iree/pull/10657) about it today. Let's rebase to main (i.e., run `git merge main`), and use the `isCompatible` method?

```suggestion
  ShapedType expectedType = inferResultType();
  if (!isCompatible(expectedType, getOutputType())) {
    return op->emitError(
               "infered type do not match provided output type. Expected ")
           << expectedType << " but got: " << getOutputType();
  }
  return success();
}
```